rapyuta_io/clients/paramserver.py

```diff
@@ -151,10 +152,16 @@ def process_dir(self, executor, rootdir, tree_path, level, dir_futures, file_fut
                     future = executor.submit(self.create_binary_file, new_tree_path, full_path)
                 if file_name.endswith('.yaml'):
                     data = parse_yaml(full_path)
-                    future = executor.submit(self.create_file, new_tree_path, data)
+                    if self.should_upload_as_binary(data, self.yaml_content_type):
```

Comment:
question: I don't quiet understand why this check is needed. The st_size check is already performed before this conditional so bigger files should already be uploaded as binary. Can you please explain?

Reply:
Refactored the code as discussed.